gumroad/includes/scanner.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Scanner
+ *
+ * The Scanner scans the post_content for gumroad links and then
+ * automatically include gumroad.js if it finds one.
+ */
+ function gum_post_scanner( $post_object ) {
+	 scanner($post_object->post_content);
+ }
+
+ /**
+  * Function accepts the post_content and scans it for gumroad short links
+	*
+	* @snce 1.2.1
+	*
+  */
+ function scanner( $post_content ) {
+ 	// counts to see if we find any gumroad product links.
+ 	if (count(preg_grep('/(gumroad.com\/l\/[[:alnum:]]|gum.co\/[[:alnum:]])/', explode("\n", $post_content))) > 0) {
+		gum_load_js('overlay');
```

Comment:
if they're using our embed, this won't work i think. we may need to load the embed here too.

Reply:
I'll make it scan for embed code like this:
```
<div class="gumroad-product-embed" data-gumroad-product-id="qLlJJ"><a href="https://gumroad.com/l/qLlJJ">Loading...</a></div>
```
This copies the short code functionality for embeds. The Gumroad interface provides the following code:
```
<script src="https://gumroad.com/js/gumroad-embed.js"></script>
<div class="gumroad-product-embed" data-gumroad-product-id="qLlJJ"><a href="https://gumroad.com/l/qLlJJ">Loading...</a></div>
```
I'll also add some guidance to the documentation to show that this is possible.

Reply:
perfect that sounds good. u rule!

Reply:
I made these changes and I think this pull request is ready.